src/java.desktop/share/classes/javax/swing/plaf/metal/MetalLookAndFeel.java

```diff
@@ -782,6 +807,8 @@ protected void initComponentDefaults(UIDefaults table) {
                           "SPACE", "pressed",
                  "released SPACE", "released"
               }),
+            // margin is (2, 14, 2, 14) which is vastly larger in horizontal
```

Comment:
I still suggest adding a reference to `BasicBorder` class or `BasicLookAndFeel` where this border is defined.

The borders for check boxes and radio buttons reference `BasicBorders.RadioButtonBorder` which helps to understand where the values are coming from.

Specifically, `BasicLookAndFeel` defines `Button.margin` property:

https://github.com/openjdk/jdk/blob/1cd488436880b00c55fa91f44c115999cf686afd/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicLookAndFeel.java#L729

It is set to the button in `installDefaults`:

https://github.com/openjdk/jdk/blob/1cd488436880b00c55fa91f44c115999cf686afd/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicButtonUI.java#L158-L160

`BasicBorders.getButtonBorder` returns a border which defines colours and margins; the margins come from `MarginBorder` class which just returns the margins of a button component:

https://github.com/openjdk/jdk/blob/1cd488436880b00c55fa91f44c115999cf686afd/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicBorders.java#L493-L495

Reply:
Agreed. This is what I found as well. Had a bit of trouble figuring out how to phrase the note since the comment was meant to differentiate Metal from the other default installed LAFs.

I moved it to `BasicLookAndFeel.java` since that's where the values are specifically defined.

Reply:
> I moved it to `BasicLookAndFeel.java` since that's where the values are specifically defined.

Perhaps, we need a note in *both files*.

In `MetalLookAndFeel.java` to refer to `BasicLookAndFeel` where the margin is defined, similar to what's done for check boxes and radio button.

And in `BasicLookAndFeel.java` to note that the above margins are vastly different from other L&Fs.

What about this text?

**`MetalLookAndFeel.java`**
```java
            // default margin is (2, 14, 2, 14), defined in
            // BasicLookAndFeel via "Button.margin" UI property.
```

**`BasicLookAndFeel.java`**
```java
            // The above margin has vastly larger horizontal values when
            // compared to other look and feels that don't rely on these values
```

Reply:
The above looks clearer. Adding a additional note in MetalLookAndFeel.java as above gives a quick lookup.

Reply:
That does sound best. Added the notes to both areas as suggested. I still don't know how to not make the note in `MetalLookAndFeel.java` not feel like it's hanging, but I added the note to the same spot again. I think it's the most logical area.

Reply:
LGTM, copyright years need to be updated for the files.